### PR TITLE
4.26.1 merge

### PR DESCRIPTION
**Pull request by simonsarginson**
GDK changes for 4.26.1
All drawn from fixes already in master.
- CI script fixes
- Test fix to deal with unreal quirks

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp [hunk 1]
```diff
@@ -151,12 +151,20 @@ void ASpatialTestRepNotify::PrepareTest()
 			}
 
 			// We consciously differ from native UE here
-			if (GetNetDriver()->IsA(USpatialNetDriver::StaticClass()))
+			// Also, the native behaviour changed when going from 4.25 to 4.26.
+			// On older versions, we expect the old array to have 3 elements, but on Spatial and on native starting from 4.26, we expect 2 elements.
+#if ENGINE_MINOR_VERSION >= 26
```
Comment (simonsarginson) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp:156]: We need Miron's future fix for this test to pass

File: ci/build-project.ps1 [hunk 2]
```diff
@@ -15,7 +15,7 @@ param(
 Write-Output "Downloading the testing project from $($test_repo_url)"
 git clone -b "$test_repo_branch" "$test_repo_url" "$test_repo_path" --depth 1
 if (-Not $?) {
-    Throw "Failed to clone testing project from $test_repo_url."
```
Comment (simonsarginson) [inline ci/build-project.ps1:18]: Thinking about it we probably don't need any of the powershell changes just yet. Even though some of this is broken

File: ci/build-project.ps1 [hunk 3]
```diff
@@ -12,10 +12,10 @@ param(
 )
 
 # Clone the testing project
-Write-Output "Downloading the testing project from $($test_repo_url)"
+Write-Output "Downloading the testing project from url: ${test_repo_url}, branch: ${test_repo_branch} to path: ${test_repo_path}."
```
Comment (simonsarginson) [inline ci/build-project.ps1:15]: improved diagnostics

File: ci/setup-build-test-gdk.ps1 [hunk 4]
```diff
@@ -6,6 +6,7 @@ param(
     [string] $unreal_engine_symlink_dir = "$build_home\UnrealEngine",
     [string] $gyms_version_path = "$gdk_home\UnrealGDKTestGymsVersion.txt"
 )
+. "$PSScriptRoot\common.ps1"
```
Comment (simonsarginson) [inline ci/setup-build-test-gdk.ps1:9]: not strictly moving this up meant no more silent failures causing the build to succeed but no tests to run.

File: ci/setup-build-test-gdk.ps1 [hunk 5]
```diff
@@ -27,12 +28,12 @@ class TestProjectTarget {
         $testing_repo_heads = git ls-remote --heads $test_repo_url $gdk_branch
         $test_gym_version = if (Test-Path -Path $test_gyms_version_path) {[System.IO.File]::ReadAllText($test_gyms_version_path)} else {[string]::Empty}
         if (Test-Path $test_env_override) {
-            $this.test_repo_branch = $test_env_override
+            $this.test_repo_branch = (Get-Item $test_env_override).value
```
Comment (simonsarginson) [inline ci/setup-build-test-gdk.ps1:31]: two fixes Miron put in that are required for CI to work for test gyms

File: ci/setup-build-test-gdk.ps1 [hunk 6]
```diff
@@ -67,8 +68,8 @@ class TestSuite {
 [string] $user_cmd_line_args = "$env:TEST_ARGS"
 [string] $gdk_branch = "$env:BUILDKITE_BRANCH"
 
-[TestProjectTarget] $gdk_test_project = [TestProjectTarget]::new("git@github.com:spatialos/UnrealGDKTestGyms.git", $gdk_branch, "Game\GDKTestGyms.uproject", "GDKTestGyms", $gyms_version_path, $env:TEST_REPO_BRANCH)
-[TestProjectTarget] $native_test_project = [TestProjectTarget]::new("git@github.com:improbable/UnrealGDKEngineNetTest.git", $gdk_branch, "Game\EngineNetTest.uproject", "NativeNetworkTestProject", $gyms_version_path, $env:NATIVE_TEST_REPO_BRANCH)
+[TestProjectTarget] $gdk_test_project = [TestProjectTarget]::new("git@github.com:spatialos/UnrealGDKTestGyms.git", $gdk_branch, "Game\GDKTestGyms.uproject", "GDKTestGyms", $gyms_version_path, "env:TEST_REPO_BRANCH")
+[TestProjectTarget] $native_test_project = [TestProjectTarget]::new("git@github.com:improbable/UnrealGDKEngineNetTest.git", $gdk_branch, "Game\EngineNetTest.uproject", "NativeNetworkTestProject", $gyms_version_path, "env:NATIVE_TEST_REPO_BRANCH")
```
Comment (simonsarginson) [inline ci/setup-build-test-gdk.ps1:72]: related to env var fix below

**Review by improbable-valentyn — APPROVED** [review]

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp:156]: You mean for it to pass on 4.25?

**Pull request closed without merging**